code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -0,0 +1,135 @@
+// ***************************************
+// *********** Nightfall
+// ***************************************
+
+/datum/action/xeno_action/activable/nightfall
+	name = "Nightfall"
+	action_icon_state = "nightfall"
+	ability_name = "Nightfall"
+	mechanics_text = "Shut down all electrical lights for 5 seconds"
+	cooldown_timer = 1 MINUTES
+	plasma_cost = 100
+	/// How far nightfall will have an effect
+	var/range = 7
+	/// How long till the lights go on again
+	var/duration = 5 SECONDS
+
+/datum/action/xeno_action/activable/nightfall/on_cooldown_finish()
+	to_chat(owner, "<span class='notice'>We gather enough mental strength to shut down lights again.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/nightfall/use_ability()
+	playsound(owner, 'sound/magic/nightfall.ogg', 50, 1)
+
+	for(var/atom/light AS in GLOB.lights)
+		light.turn_light(null, FALSE, duration, TRUE, TRUE, get_turf(owner), range)
```

Comment:
this is unperformant and should use a range check instead

Reply:
the range check is done by turn light

Reply:
it shouldnt be why is it